### PR TITLE
timerdrift:stop ntp service before test in guest

### PR DESCRIPTION
**Pull request by maxujun**
Signed-off-by: Xujun Ma <xuma@redhat.com>
ID:1708217

### CODE REVIEW

**Comment by maxujun** [comment]
@MiriamDeng @YanhuiMa  Could you help review?

**Comment by YanhuiMa** [comment]
Hi xujun, for x86 we don't use the auto case now, because the auto case is not the same with polarion case steps. If you want to cover the related cases, you can use https://github.com/autotest/tp-qemu/pull/2204.
If you just want to fix the auto bug, please provide the test results.

**Comment by MiriamDeng** [comment]
> Hi xujun, for x86 we don't use the auto case now, because the auto case is not the same with polarion case steps. If you want to cover the related cases, you can use #2204.
Hi xujun,
Is it ppc only test case now ? 
Thanks
Min

**Comment by yanan-fu** [comment]
@maxujun  Any new update for this patch now ? Thanks!

**Comment by maxujun** [comment]
(1/2) Host_RHEL.m8.u5.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_stop.clock_host: PASS (182.08 s)
 (2/2) Host_RHEL.m8.u5.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_stop.clock_vm: PASS (181.24 s)

**Comment by maxujun** [comment]
> 
> 
> > Hi xujun, for x86 we don't use the auto case now, because the auto case is not the same with polarion case steps. If you want to cover the related cases, you can use #2204.
> > Hi xujun,
> > Is it ppc only test case now ?
> > Thanks
> > Min
yes

File: qemu/tests/timedrift_with_stop.py [hunk 1]
```diff
@@ -53,7 +53,10 @@ def run(test, params, env):
     stop_iterations = int(params.get("stop_iterations", 1))
     stop_time = int(params.get("stop_time", 60))
     stop_with_signal = params.get("stop_with_signal") == "yes"
+    ntp_stop_cmd = params["ntp_stop_cmd"]
 
+    logging.info("Stop ntp service in guest")
+    session.cmd(ntp_stop_cmd)
```
Comment (MiriamDeng) [inline qemu/tests/timedrift_with_stop.py:59]: it had better add a check point for this command ?

Comment (MiriamDeng) [reply qemu/tests/timedrift_with_stop.py:59]: Test result

Host_RHEL.m8.u5.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_stop.clock_host: PASS
Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_stop.clock_vm: PASS

**Comment by YanhuiMa** [comment]
Hi xujun, could you tell me the reason you run the case for ppc? AFAK, the auto case does match with our polarion cases steps?

**Comment by maxujun** [comment]
> 
> 
> Hi xujun, could you tell me the reason you run the case for ppc? AFAK, the auto case does match with our polarion cases steps?

This case was also included in timer_device loop file for x86.Could you help confirm if it's be using for x86?

**Comment by YanhuiMa** [comment]
> > Hi xujun, could you tell me the reason you run the case for ppc? AFAK, the auto case does match with our polarion cases steps?
> 
> This case was also included in timer_device loop file for x86.Could you help confirm if it's be using for x86?

Yes, it is still in x86 test loop, I will remove it later. Thanks for reminding that.

**Comment by yanan-fu** [comment]
> Hi xujun, could you tell me the reason you run the case for ppc? AFAK, the auto case does match with our polarion cases steps?

@maxujun  Could you explain more about this issue above ?  Is there any ppc specific reason for using this case instead of the https://github.com/autotest/tp-qemu/pull/2204 ? I wonder which is the expected auto script for the manual case. cc @YanhuiMa

**Comment by maxujun** [comment]
(1/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_migration: PASS (215.69 s)
 (2/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_stop.clock_host: PASS (171.75 s)
 (3/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.shared_ntp_date.with_stop.clock_vm: PASS (172.74 s)
 (4/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.timedrift.with_cpu_offline: PASS (759.54 s)

**Comment by maxujun** [comment]
Updated,fix the same problem for two other cases timedrift.shared_ntp_date.with_migration and timedrift.with_cpu_offline.
@YanhuiMa Did you use these two case?

**Comment by maxujun** [comment]
> 
> 
> > Hi xujun, could you tell me the reason you run the case for ppc? AFAK, the auto case does match with our polarion cases steps?
> 
> @maxujun Could you explain more about this issue above ? Is there any ppc specific reason for using this case instead of the #2204 ? I wonder which is the expected auto script for the manual case. cc @YanhuiMa
#2204 have problem for ppc,I have file another patch to fix them, will use them in the future.but this case actually have problem,need fix include other two cases  timedrift.shared_ntp_date.with_migration and timedrift.with_cpu_offline.

**Comment by YanhuiMa** [comment]
> Updated,fix the same problem for two other cases timedrift.shared_ntp_date.with_migration and timedrift.with_cpu_offline.
> @YanhuiMa Did you use these two case?

yes, now they are in test loop.

**Comment by maxujun** [comment]
> 
> 
> > Updated,fix the same problem for two other cases timedrift.shared_ntp_date.with_migration and timedrift.with_cpu_offline.
> > @YanhuiMa Did you use these two case?
> 
> yes, now they are in test loop.

Could you help review this patch? I have updated them.

File: generic/tests/cfg/timedrift.cfg [hunk 2]
```diff
@@ -3,6 +3,7 @@
     qemu_stop = off
     i386, x86_64:
         rtc_drift = "slew"
+    ntp_stop_cmd = "(systemctl stop chronyd || service ntpdate stop)"
```
Comment (yanan-fu) [inline generic/tests/cfg/timedrift.cfg:6]: window os need be considered too.

Comment (maxujun) [reply generic/tests/cfg/timedrift.cfg:6]: I don't know if have this problem on x86 platform, @YanhuiMa Could you help test these cases on x86?

Comment (YanhuiMa) [reply generic/tests/cfg/timedrift.cfg:6]: You mean what problem? For the case, it indeed needs to add ntp_stop_cmd for both x86 and ppc. And the case is not specified only ppc. So yfu and I think cmd related window guest also need to be considered.

**Comment by maxujun** [comment]
Close this patch,use other cases instead of.

**Pull request closed without merging**